docs/docs/models-definition.md

```diff
@@ -1,25 +1,30 @@
 ## Definition
 
-To define mappings between a model and a table, use the `define` method. Sequelize will then automatically add the attributes `createdAt` and `updatedAt` to it. So you will be able to know when the database entry went into the db and when it was updated the last time. If you do not want timestamps on your models, only want some timestamps, or you are working with an existing database where the columns are named something else, jump straight on to [configuration ][0]to see how to do that.
+To define mappings between a model and a table in v4, you define a class that inherits from [`Sequelize.Model`](../api/model) and call [`Model.init(attributes, options)`](../api/model#init). The model name is determined by the class name. You need to pass the sequelize instance as `options.sequelize` (see example)
+Sequelize will then automatically add the attributes `createdAt` and `updatedAt` to it. So you will be able to know when the database entry went into the db and when it was updated the last time. If you do not want timestamps on your models, only want some timestamps, or you are working with an existing database where the columns are named something else, jump straight on to [configuration ][0]to see how to do that.
 
+Defining a model with [`sequelize.define()`](../api/sequelize#define) as in previous versions is still supported. Check the v3 docs for examples.
 
 ```js
-var Project = sequelize.define('project', {
+class Project extends Sequelize.Model { }
+Project.init({
   title: Sequelize.STRING,
   description: Sequelize.TEXT
-})
+}, { sequelize })
```

Comment:
> normalization can happen just in time 

I don't think this is good because it surely has a performance impact if we either don't normalize and adapt every codepiece where options are accessed or results in way more code if we call `_normalizeIfNotNormalizedYet()` everywhere.

> As for exposing the normalization, i agree that's a tricky issue for options, less so for attributes.

I don't think so - aren't the data types normalized (e.g. if you passed `Sequelize.STRING` it will be normalized to `new Sequelize.STRING()`)?

> I would like to see something like: ...

The table name is already in `options.tableName`. But is your ES6 compat so much different from a `Model.init()` call? Because if you are using Babel/TS already, you can use decorators, which work even better. This is the API I would like to see (and this supported as of 4.0.0-1):
```ts
// TypeScript

@Options({ sequelize, tableName: ''})
class User extends Model {
  @Attribute({ type: DataTypes.STRING, unique: true })
  public username: string;
}

// ES7 / Babel
@Options({ sequelize, tableName: ''})
@Attributes({
  username: {
    type: DataTypes.STRING,
    unique: true
  }
})
class User extends Model {

}

// ES6 compat
class User extends Model {

}
Model.init({
  username: {
    type: DataTypes.STRING,
    unique: true
  }
}, {
  sequelize,
  tableName: ''
})
```

> Whatever we decide on, we definitely don't have a final API so i don't think we should advertise anything other than the original .define API untill we do.

True, the API is not final, but neither is the prerelease we already have out for v4. Which means people can already use this, but they don't have any docs on it and cannot provide feedback to the new API.